Update API docs to reflect Grants API cleanup

bellatumaneng-okta opened the pull request:
## Description:
- Accompanying changes for https://github.com/okta/okta-core/pull/28254 to clean up Grants API
- Add correct `issuer` value and remove `issuerId`
- Remove `"issuer"` and `"revoke"` HAL links and add hints for `"self"` for `GET` and `DELETE`

### Resolves:
* [OKTA-175296](https://oktainc.atlassian.net/browse/OKTA-175296)

jakubvul commented:
@bellatumaneng-okta which release is this supposed to go out with?

bellatumaneng-okta commented:
The code changes haven't merged yet because I'm waiting for feedback from Product. It'll probably be 2018.30 if I don't get approval from that side today.

danielzeller-okta commented on _source/_docs/api/resources/users.md:
```diff
@@ -4249,7 +4225,7 @@ For an individual User result, the Links Object contains a full set of link rela
 | created     | Timestamp when the grant was created                                                                                           | Date                                                            |
 | lastUpdated | Timestamp when the grant was last updated                                                                                      | Date                                                            |
-| issuerId    | ID of the authorization server for this grant                                                                                  | String                                                          |
+| issuer      | The complete URL of the authorization server for this grant                                                                    | String                                                          |
```
Is this how we describe the issuer in other places in the docs? Seems a little clunky

jakubvul replied:
We don't have a consistent way of describing this property: https://developer.okta.com/docs/api/resources/oidc

bellatumaneng-okta replied:
I used the phrasing from here: https://developer.okta.com/docs/api/resources/oidc#response-properties-6

danielzeller-okta commented on _source/_docs/api/resources/users.md:
```diff
@@ -4184,52 +4166,46 @@ For an individual User result, the Links Object contains a full set of link rela
 
 ~~~sh
 {
-    “id”: “oag2n8HU1vTmvCdQ50g3",
```
any clue why we were using these different quotation mark characters? +1 to cleanup

jakubvul replied:
Yea we should be using `"` only. None of these fancy `“` and `”`. 😄

bellatumaneng-okta commented:
FYI @jakubvul these changes should be going with this week's release. Thanks!

jakubvul commented:
@bellatumaneng-okta I don't see it in the list for this week's release: https://oktawiki.atlassian.net/wiki/spaces/eng/pages/538020070/Release+Notes+2018.29

Also, you will need to clean up the non-unicode characters before you merge this.

Attn @eklein-okta

bellatumaneng-okta commented:
@jakubvul sorry, I was referring to this week's code freeze, so 2018.30.

I _think_ I cleaned up all the funny quotes in every snippet I modified for the current changes, but if you saw something else let me know and I can double check.

jakubvul commented:
Indeed! Sorry, I misread the diff.

nbarbettini commented:
Don't merge: waiting for 2018.31
